[Security] Add documentation chapter for OAuth2TokenHandler (#22029)
symfony/symfony-docs: security/access_token.rst

        }
    }

3) Configure the Oauth2TokenHandler
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Symfony provides a generic ``Oauth2TokenHandler`` that calls the configured token introspection endpoint to validate the token and retrieve the user information from it.

Reviewer: The [Oauth2TokenHandler class](https://github.com/symfony/symfony/blob/8.1/src/Symfony/Component/Security/Http/AccessToken/OAuth2/Oauth2TokenHandler.php) is marked internal, do we want to mention it ?

> Symfony provides a generic ``Oauth2TokenHandler`` that calls the configured token introspection endpoint to validate the token and retrieve the user information from it.

Reviewer 2: ```suggestion
Symfony provides a generic access token handler that calls the configured token introspection endpoint to validate the token and retrieve the user information from it.
```

It requires the ``symfony/http-client`` package to make the needed HTTP requests. If you haven't installed it yet, run this command:

.. code-block:: terminal

    $ composer require symfony/http-client

First, configure a dedicated scoped HTTP client for the token handler:

.. configuration-block::

>     .. code-block:: yaml

Reviewer 3: ```suggestion
    .. code-block:: yaml

```

                    access_token:
                        token_handler:
                            oauth2: ~
                        token_extractors: 'header'
                        realm: 'My API'

    .. code-block:: xml

        <!-- config/packages/security.xml -->
        <?xml version="1.0" encoding="UTF-8"?>
        <srv:container xmlns="http://symfony.com/schema/dic/security"
            xmlns:srv="http://symfony.com/schema/dic/services"
            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://symfony.com/schema/dic/services
                https://symfony.com/schema/dic/services/services-1.0.xsd
                http://symfony.com/schema/dic/security
                https://symfony.com/schema/dic/security/security-1.0.xsd">

            <config>
                <firewall name="main">
                    <access-token>
                        <token-handler>
                            <oauth2/>
                        </token-handler>
                    </access-token>
                </firewall>
            </config>
        </srv:container>

    .. code-block:: php

        // config/packages/security.php

>         use Symfony\Config\SecurityConfig;
> 
>         return static function (SecurityConfig $security) {
>             $security->firewall('main')
>                 ->accessToken()
>                     ->tokenHandler()
>                         ->oauth2()
>                     ->tokenExtractors('header')
>                     ->realm('My API')
>             ;
>         };

Reviewer 3: ```suggestion
        namespace Symfony\Component\DependencyInjection\Loader\Configurator;

        return App::config([
            'security' => [
                'firewalls' => [
                    'main' => [
                        'access_token' => [
                            'token_handler' => [
                                'oauth2' => null,
                            ],
                            'token_extractors' => 'header',
                            'realm' => 'My API',
                        ],
                    ],
                ],
            ],
        ]);
```

Author: As I'm targeting 7.3, I think we should keep the "old" PHP way, and changing it in another PR targeting 8.0.
@javiereguiluz WDYT?

Reviewer: > As I'm targeting 7.3, I think we should keep the "old" PHP way, and changing it in another PR targeting 8.0. @javiereguiluz WDYT?

I agree, documentation for 7.4 is still using the fluent PHP configuration and some features documented recently have used this syntax
